File: apps/bff/src/platform/http.ts
import type { ApiError } from '@tooday/shared';
import type { Context } from 'hono';
import type { ContentfulStatusCode } from 'hono/utils/http-status';

export function errorResponse({
  c,
  status,
  code,
  message,
}: {
  c: Context;
  status: ContentfulStatusCode;
  code: string;
  message: string;
}) {
  return c.json<ApiError>({ error: { code, message } }, status);
}

File: rust/crates/bff/src/platform/http.rs
use axum::http::StatusCode;
use axum::response::{IntoResponse, Response};
use axum::Json;
use tooday_shared::{ApiError, ApiErrorBody};

pub fn error_response(status: StatusCode, code: &str, message: &str) -> Response {
    (
        status,
        Json(ApiError {
            error: ApiErrorBody { code: code.to_owned(), message: message.to_owned() },
        }),
    )
        .into_response()
}
